ckanext/ldap/lib/helpers.py

```diff
@@ -3,9 +3,17 @@
 """
 Created by 'bens3' on 2013-06-21.
 Copyright (c) 2013 'bens3'. All rights reserved.
+Amended: hvw / 2018
 """
 
 import pylons
+from ckan.plugins.toolkit import c
```

Comment:
This line errors under my test conditions because of the older CKAN version I'm using. Perhaps it would be worth adding `ckantoolkit` to avoid compatibility problems?

Reply:
Sorry, I was a bit hasty with my assessment of the issue I was seeing. I was testing against version `2.3a` which like you say should be fine. The issue I saw was:

```
... more trace omitted ...
  File "/usr/lib/ckan/default/local/lib/python2.7/site-packages/paste/deploy/util.py", line 56, in fix_call
    val = callable(*args, **kw)
  File "/usr/lib/ckan/default/src/ckan/ckan/config/middleware.py", line 58, in make_app
    load_environment(conf, app_conf)
  File "/usr/lib/ckan/default/src/ckan/ckan/config/environment.py", line 232, in load_environment
    p.load_all(config)
  File "/usr/lib/ckan/default/src/ckan/ckan/plugins/core.py", line 134, in load_all
    load(*plugins)
  File "/usr/lib/ckan/default/src/ckan/ckan/plugins/core.py", line 149, in load
    service = _get_service(plugin)
  File "/usr/lib/ckan/default/src/ckan/ckan/plugins/core.py", line 255, in _get_service
    return plugin.load()(name=plugin_name)
  File "/usr/lib/ckan/default/local/lib/python2.7/site-packages/pkg_resources.py", line 2048, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "/usr/lib/ckan/default/src/ckanext-ldap/ckanext/ldap/plugin.py", line 10, in <module>
    from ckanext.ldap.lib.helpers import is_ldap_user, get_login_action
  File "/usr/lib/ckan/default/src/ckanext-ldap/ckanext/ldap/lib/helpers.py", line 10, in <module>
    from ckan.plugins.toolkit import c
ImportError: cannot import name c
```

This doesn't need to be fixed with `ckantoolkit`, it can be resolved by using `from ckan.plugins import toolkit` and then referencing `c` as `toolkit.c` as required. I assume this is an ordering and lazy initialisation issue.

Reply:
OK. Did that, hope it rids you of the error (which I don't fully understand, to be honest).